server/src/main/java/org/elasticsearch/action/search/QueryPhaseResultConsumer.java

```diff
@@ -352,19 +352,18 @@ private synchronized void onMergeFailure(Exception exc) {
                 circuitBreakerBytes = 0;
             }
             failure.compareAndSet(null, exc);
-            MergeTask task = runningTask.get();
-            runningTask.compareAndSet(task, null);
-            onPartialMergeFailure.accept(exc);
-            List<MergeTask> toCancels = new ArrayList<>();
+            final MergeTask task = runningTask.getAndSet(null);
+            final List<Releasable> toCancels = new ArrayList<>();
```

Comment:
Does it make sense to name the list `releasables` instead of `toCancels`? It sounds to me more logical because its element implement `Relesable`, not `Closeable`